app/src/main/java/com/example/reduxtestapp/ui/country/CountryViewState.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.reduxtestapp.ui.country
+
+import com.example.reduxtestapp.data.model.country.asPresentation
+import com.example.reduxtestapp.redux.AppState
+import com.example.reduxtestapp.redux.CountryState
+
+data class CountryViewState (
+    val countryList: List<CountryUiData> = listOf(),
+    val status: CountryState.Status = CountryState.Status.SUCCESS,
+)
+
+fun AppState.toCountryViewState() = CountryViewState(
+    countryList = countryState.countryList.asPresentation(),
+    status = countryState.status
+)
+
+data class CountryUiData (
+    val name: String,
+    val languages: HashMap<String, String>
```

Comment:
declare as `Map` instead of implementation

also a general naming comment; you can drop `Ui` and `Data` from class names and name them something that represents what their role is or how they're presented on the screen in case of view states - maybe `CountryItem` or just `Country` would be ok